### PR TITLE
Make docs processing more robust

### PR DESCRIPTION
**Pull request by felixfontein**
Currently the devel docs build fails with https://pastebin.com/K1u95BXx.

### CODE REVIEW

File: src/antsibull_docs/cli/doc_commands/stable.py [hunk 1]
```diff
@@ -166,7 +169,8 @@ def normalize_plugin_info(plugin_type: str,
             # But we can use the default value (some variant of "empty") for everything else
             # Note: We looped through doc first and returned an exception if doc did not normalize
             # so we're able to use it in the error message here.
-            errors.append(f'Unable to normalize {new_info["doc"]["name"]}: {field}'
+            plugin_name = (new_info["doc"] or {}).get("name") or "???"
```
Comment (briantist) [inline src/antsibull_docs/cli/doc_commands/stable.py:172]: I would appreciate a comment in the code explaining how/why the string `"???"` could end up here

**Review by briantist — APPROVED** [review]
Just a non-blocking request for a comment on the confusing  `???` value

File: src/antsibull_docs/cli/doc_commands/stable.py (hunk 1 above)
Comment (felixfontein) [reply src/antsibull_docs/cli/doc_commands/stable.py]: I added a comment in 3b0a9328290d1138d386d09feb3ae78601d7beee, and at the same time decided to rewrite that function a bit.

**Review by briantist — APPROVED** [review]

**Comment by felixfontein** [comment]
Hmm, something isn't right yet. Need to look at this some more.

File: src/antsibull_docs/cli/doc_commands/stable.py (hunk 1 above)
Comment (felixfontein) [reply src/antsibull_docs/cli/doc_commands/stable.py]: Actually I undid most changes to this function.

**Comment by felixfontein** [comment]
I'm now happy with the actual docsite output.

(This project definitely needs better tests - probably we need to include a test collection and the expected RST output for it.)

**Comment by felixfontein** [comment]
@briantist thanks for reviewing this!

**Pull request merged**